### ISSUE TITLE
encryption config does not have attribute "provider_key_arn"

### ISSUE THREAD

**Issue by aiell0** [terraform-aws-modules/terraform-aws-eks#2989]
## Description

When setting `create_kms_key` to false, and also not specifying a `cluster_encryption_config`, I am getting the following:

```
│ Error: Unsupported attribute
│
│   on .terraform/modules/eks.eks_bottlerocket/main.tf line 80, in resource "aws_eks_cluster" "this":
│   80:         key_arn = var.create_kms_key ? module.kms.key_arn : encryption_config.value.provider_key_arn
│     ├────────────────
│     │ encryption_config.value is object with 1 attribute "resources"
│
│ This object does not have an attribute named "provider_key_arn".
```

Before you submit an issue, please perform the following first:

1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/`
2. Re-initialize the project root to pull down modules: `terraform init`
3. Re-attempt your terraform plan or apply and check if the issue still persists

## Versions

- Module version [Required]: ~>20.0

- Terraform version: 1.7.5
- Provider version(s): ~>5.0

## Reproduction Code [Required]

```
module "eks_bottlerocket" {
  source  = "terraform-aws-modules/eks/aws"
  version = "~>20.0"

  cluster_name    = var.cluster_name
  cluster_version = var.cluster_version
  create_kms_key  = var.create_kms_key

  # TODO: Find a more secure way to access the cluster
  cluster_endpoint_public_access = true

  # EKS Addons
  cluster_addons = {
    # aws-ebs-csi-driver = {}
    coredns    = {}
    kube-proxy = {}
    vpc-cni    = {}
  }

  vpc_id     = var.vpc_id
  subnet_ids = var.subnet_ids

  eks_managed_node_groups = {
    bottlerocket = {
      ami_type = "BOTTLEROCKET_x86_64"
      platform = "bottlerocket"

      instance_types = ["m6i.large", "m5.large", "m5n.large", "m5zn.large"]
      capacity_type  = "SPOT"
      disk_size      = 50

      min_size     = 1
      max_size     = 3
      desired_size = 2
    }
  }

  # Cluster access entry
  enable_cluster_creator_admin_permissions = false

  access_entries = {
    terraform = {
      kubernetes_groups = []
      principal_arn     = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/terraform-cloud-role"

      policy_associations = {
        admin = {
          policy_arn = "arn:aws:eks::aws:cluster-access-policy/AmazonEKSClusterAdminPolicy"
          access_scope = {
            type = "cluster"
          }
        }
      }
    }
    devops = {
      kubernetes_groups = []
      principal_arn     = tolist(data.aws_iam_roles.devops.arns)[0]

      policy_associations = {
        admin = {
          policy_arn = "arn:aws:eks::aws:cluster-access-policy/AmazonEKSAdminPolicy"
          access_scope = {
            type = "cluster"
          }
        }
      }
    }
    developers = {
      kubernetes_groups = []
      principal_arn     = tolist(data.aws_iam_roles.developers.arns)[0]

      policy_associations = {
        admin = {
          policy_arn = "arn:aws:eks::aws:cluster-access-policy/AmazonEKSViewPolicy"
          access_scope = {
            type = "cluster"
          }
        }
      }
    }
  }
}
```

## Expected behavior

Terraform plan goes through successfully. Selects the default KMS key if not specified.

## Actual behavior

Terraform plan fails.

### Terminal Output Screenshot(s)

```
│ Error: Unsupported attribute
│
│   on .terraform/modules/eks.eks_bottlerocket/main.tf line 80, in resource "aws_eks_cluster" "this":
│   80:         key_arn = var.create_kms_key ? module.kms.key_arn : encryption_config.value.provider_key_arn
│     ├────────────────
│     │ encryption_config.value is object with 1 attribute "resources"
│
│ This object does not have an attribute named "provider_key_arn".
```

## Additional context

I'm using Terraform Cloud if that helps. I also didn't include all of my code, just the relevant bits.

**Comment by aiell0** [comment]
This can be fixed by setting `cluster_encryption_config = {}` but I feel that's an interface bug as that's not very intuitive.

**Comment by bryantbiggs** [comment]
its a two part problem, so two arguments are required. by default we want to encrypt secrets, and that requires a key. therefore, we also default to creating a key for users. if you wish to back completely out of encryption, that will require the two arguments listed. 

tl;dr - the not recommended path is the harder path

**Comment by aiell0** [comment]
Fair enough - feel free to close it then.
